docker/go-control-plane/controlplane/resource.go

```diff
@@ -0,0 +1,139 @@
+func MakeCluster(clusterName, upstreamHost string, upstreamPort uint32) *pb_cluster.Cluster {
+	return &pb_cluster.Cluster{
+		Name:                 clusterName,
+		ConnectTimeout:       durationpb.New(5 * time.Second),
+		ClusterDiscoveryType: &pb_cluster.Cluster_Type{Type: pb_cluster.Cluster_LOGICAL_DNS},
+		LbPolicy:             pb_cluster.Cluster_ROUND_ROBIN,
+		LoadAssignment:       makeEndpoint(clusterName, upstreamHost, upstreamPort),
+		DnsLookupFamily:      pb_cluster.Cluster_V4_ONLY,
+	}
+}
+
+func makeEndpoint(clusterName, upstreamHost string, upstreamPort uint32) *pb_endpoint.ClusterLoadAssignment {
+	return &pb_endpoint.ClusterLoadAssignment{
+		ClusterName: clusterName,
+		Endpoints: []*pb_endpoint.LocalityLbEndpoints{{
+			LbEndpoints: []*pb_endpoint.LbEndpoint{{
+				HostIdentifier: &pb_endpoint.LbEndpoint_Endpoint{
+					Endpoint: &pb_endpoint.Endpoint{
+						Address: &pb_core.Address{
+							Address: &pb_core.Address_SocketAddress{
+								SocketAddress: &pb_core.SocketAddress{
+									Protocol: pb_core.SocketAddress_TCP,
+									Address:  upstreamHost,
+									PortSpecifier: &pb_core.SocketAddress_PortValue{
+										PortValue: upstreamPort,
+									},
+								},
+							},
+						},
+					},
+				},
+			}},
+		}},
+	}
+}
+
+// MakeHTTPListener builds a LDS resource that routes traffic to a given
+// cluster.
+func MakeHTTPListener(listenerName, cluster string) *pb_listener.Listener {
+	any_route, _ := anypb.New(&pb_router.Router{})
+	httpcm := &pb_hcm.HttpConnectionManager{
+		RouteSpecifier: &pb_hcm.HttpConnectionManager_RouteConfig{
+			RouteConfig: &pb_route.RouteConfiguration{
+				VirtualHosts: []*pb_route.VirtualHost{
+					{
+						Domains: []string{"*"},
+						Routes: []*pb_route.Route{
+							{
+								Match: &pb_route.RouteMatch{
+									PathSpecifier: &pb_route.RouteMatch_Prefix{},
+								},
+								Action: &pb_route.Route_Route{
+									Route: &pb_route.RouteAction{
+										ClusterSpecifier: &pb_route.RouteAction_Cluster{
+											Cluster: cluster,
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+		HttpFilters: []*pb_hcm.HttpFilter{
+			{
+				Name: "router",
+				ConfigType: &pb_hcm.HttpFilter_TypedConfig{
+					TypedConfig: any_route,
+				},
+			},
+		},
+	}
+	any_hcm, err := anypb.New(httpcm)
+	if err != nil {
+		panic(err)
+	}
+	return &pb_listener.Listener{
+		Name: listenerName,
+		ApiListener: &pb_listener.ApiListener{
+			ApiListener: any_hcm,
+		},
+	}
+}
+
+// GenerateSnapshot prepares a new xDS config snapshot for serving to clients.
+func GenerateSnapshot(upstreamHost string, upstreamPort uint32) *cache.Snapshot {
+	snap, _ := cache.NewSnapshot(strconv.Itoa(snapshot_version), map[resource.Type][]types.Resource{
+		resource.ClusterType:  {MakeCluster(ClusterName, upstreamHost, upstreamPort)},
+		resource.ListenerType: {MakeHTTPListener(ListenerName, ClusterName)},
+	})
+	snapshot_version++
+	return snap
+}
```

Comment:
This function is not used anymore. Please delete it.

Reply:
Done